explorer/ast/static_scope.h

```diff
@@ -35,11 +36,18 @@ class StaticScope {
   };
 
   // Construct a root scope.
-  StaticScope() = default;
+  explicit StaticScope(Nonnull<TraceStream*> trace_stream)
+      : ast_node_(std::nullopt), trace_stream_(trace_stream){};
 
   // Construct a scope that is nested within the given scope.
-  explicit StaticScope(Nonnull<const StaticScope*> parent)
-      : parent_scope_(parent) {}
+  explicit StaticScope(
+      Nonnull<const StaticScope*> parent,
+      std::optional<Nonnull<const AstNode*>> ast_node = std::nullopt)
```

Comment:
Can we make the `ast_node` parameter mandatory?

Reply:
Yes we can do that. I'll do it in the next commit.